include/spblas/algorithms/algorithms.hpp

```diff
@@ -1,10 +1,14 @@
 #pragma once
 
 #include <spblas/algorithms/scale.hpp>
+
 #include <spblas/algorithms/scale_impl.hpp>
 
-#include <spblas/algorithms/multiply.hpp>
+// #include <spblas/algorithms/multiply.hpp>
+
+#ifndef ENABLE_ONEMKL
 #include <spblas/algorithms/multiply_impl.hpp>
```

Comment:
should this have an "mkl" namespace in the path for this include ?

Reply:
oh wait, this is the default one, not the mkl one right ?

Reply:
Correct.  What happens right now is if the oneMKL backend is enabled, then the oneMKL backend will include its own `multiply_impl.hpp` with implementations of multiply.  Only if there is no vendor backend do we need to include my reference implementation.

There's a broader question here of what to do about `spblas/algorithms/multiply.hpp`, which defines the API.  If we leave it included as written, there's actually a *linker error* if the backend doesn't have an implementation of multiply for the combination the user has given.  (Suppose they try to multiply a COO times a CSC and the vendor backend has no implementation of that.)  I've left it commented out here so that there's actually a compile-time error instead of a linker error.

There are two potential ways to do this:

1) `multiply` and other functions are not forward declared at all.  There are only the vendor declared implementations, and users will get errors from them based on what vendors decide.  This is what currently happens, and it gives the vendors maximum power.
2) Put the vendor implementations inside a `__vendor` namespace and have a standard implementation of `multiply` try to call in to them.  `multiply` could then give a nice error that there is no vendor implementation of that particular invocation if the call to the `__vendor::multiply` would fail.  (Of course vendors could do this themselves with 1), but this would avoid duplicating the error infrastructure.)